tests/unit/test_client.py

```diff
@@ -1106,3 +1106,61 @@ def test_verify_takes_precedence_over_scoped_config(self):
                               scoped_config={'ca_bundle': 'mycerts.pem'})
         call_kwargs = self.endpoint_creator.create_endpoint.call_args[1]
         self.assertEqual(call_kwargs['verify'], 'verify.pem')
+
+
+class TestConfig(unittest.TestCase):
+    def test_merge_returns_new_config_object(self):
+        config = client.Config()
+        other_config = client.Config()
+        new_config = config.merge(other_config)
+        # Check the type is correct
+        self.assertIsInstance(new_config, client.Config)
+        # Make sure the new config is a brand new config object
+        self.assertIsNot(new_config, config)
+        self.assertIsNot(new_config, other_config)
+
+    def test_general_merge_keeps_default_values(self):
+        config = client.Config()
+        other_config = client.Config()
+        config_properties = vars(config)
+        new_config = config.merge(other_config)
+        # Ensure that the values all stayed the same in the new config
+        self.assertEqual(config_properties, vars(new_config))
+
+    def test_merge_overrides_values(self):
+        config = client.Config(region_name='us-east-1')
+        other_config = client.Config(region_name='us-west-2')
+        new_config = config.merge(other_config)
+        self.assertEqual(new_config.region_name, 'us-west-2')
+
+    def test_merge_overrides_only_when_non_default_values(self):
+        config = client.Config(
+            region_name='us-east-1', signature_version='s3v4')
+        other_config = client.Config(region_name='us-west-2')
+        new_config = config.merge(other_config)
+        self.assertEqual(new_config.region_name, 'us-west-2')
+        self.assertEqual(new_config.signature_version, 's3v4')
+
+    def test_merge_default_does_not_override_connect_timeout(self):
+        config = client.Config(connect_timeout=55)
+        other_config = client.Config()
+        new_config = config.merge(other_config)
+        self.assertEqual(new_config.connect_timeout, 55)
+
+    def test_merge_override_connect_timeout(self):
+        config = client.Config(connect_timeout=55)
+        other_config = client.Config(connect_timeout=75)
+        new_config = config.merge(other_config)
+        self.assertEqual(new_config.connect_timeout, 75)
+
+    def test_merge_default_does_not_override_read_timeout(self):
```

Comment:
Why is this test needed given `test_merge_default_does_not_override_connect_timeout`?